api/internal/plugins/fnplugin/fnplugin.go

```diff
@@ -51,13 +51,16 @@ func resourceToRNode(res *resource.Resource) (*yaml.RNode, error) {
 }
 
 // GetFunctionSpec return function spec is there is. Otherwise return nil
-func GetFunctionSpec(res *resource.Resource) *runtimeutil.FunctionSpec {
+func GetFunctionSpec(res *resource.Resource) (*runtimeutil.FunctionSpec, error) {
 	rnode, err := resourceToRNode(res)
 	if err != nil {
-		return nil
+		return nil, fmt.Errorf("%w", err)
```

Comment:
```suggestion
		return nil, err
```

Reply:
🤔 Actually, it would probably be better to add a prefix here

So maybe something like `return nil, fmt.Errorf("could not convert resource to RNode: %w", err)`

Reply:
done